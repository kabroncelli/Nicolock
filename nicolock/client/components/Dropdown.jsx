import React from 'react'
import PropTypes from 'prop-types'

class Dropdown extends React.Component {
  static propTypes = {
    valueEntered: PropTypes.func.isRequired,
    options: PropTypes.array.isRequired,
    label: PropTypes.string.isRequired,
  }

  constructor (props) {
    super(props)
    let initialRadius
    if (props.options.length > 0) {
      initialRadius = props.options[0]
      props.valueEntered(initialRadius)
    }
    this.state = {
      value: initialRadius || '',
    }
  }

  valueEntered = (e) => {
    let newText
    try {
      newText = parseInt(e.target.value)
    } catch (err) {
      return
    }
    this.setState({value: newText})
    this.props.valueEntered(newText)
  }

  renderOptions () {
    return (
      this.props.options.map(option =>
        <option key={option} value={option}>{option}</option>
      )
    )
  }

  render () {
    return (
      <span>
        <label className='label'>{this.props.label}</label>
        <select className='react-dropdown' onChange={this.valueEntered}>
          {this.renderOptions()}
        </select>
      </span>
    )
  }
}

export default Dropdown
