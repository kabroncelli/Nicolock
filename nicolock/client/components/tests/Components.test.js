import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Dropdown from '../Dropdown'

configure({ adapter: new Adapter() })

describe('Components render correctly', () => {
  it('Dropdown renders correctly', () => {
    const valueEntered = jest.fn()
    const wrapper = shallow(
      <Dropdown
        options={[1, 2, 3]}
        label='dropdown test'
        valueEntered={valueEntered} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
