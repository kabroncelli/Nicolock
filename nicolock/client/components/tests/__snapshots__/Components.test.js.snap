// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components render correctly Dropdown renders correctly 1`] = `
<span>
  <label
    className="label"
  >
    dropdown test
  </label>
  <select
    className="react-dropdown"
    onChange={[Function]}
  >
    <option
      key="1"
      value={1}
    >
      1
    </option>
    <option
      key="2"
      value={2}
    >
      2
    </option>
    <option
      key="3"
      value={3}
    >
      3
    </option>
  </select>
</span>
`;
